### PR TITLE
[serve] Remove v1 api

### PR DESCRIPTION
**Pull request by zcin**
## Why are these changes needed?

Remove:
- `deployment.deploy()`
- `deployment.delete()`
- `deployment.get_handle()`
- `serve.get_deployment()`
- `serve.list_deployments()`

Temporarily, our tests are unchanged and have been moved to:
- `deployment.deploy()` -> `deployment._deploy()`
- `deployment.delete()` -> `deployment._delete()`
- `deployment.get_handle()` -> `deployment._get_handle()`
- `serve.get_deployment()` -> `_private_api.get_deployment()`
- `serve.list_deployments()` -> `_private_api.list_deployments()`

I will follow up this PR with PR(s) to remove or migrate our V1 tests.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by edoakes — APPROVED** [review]

**Comment by edoakes** [comment]
@zcin I think the workflow test failures are related

**Review by sihanwang41 — APPROVED** [review]

File: python/ray/serve/api.py [hunk 1]
```diff
@@ -426,37 +426,13 @@ def decorator(_func_or_class):
...
-    Args:
-        name: name of the deployment. This must have already been
-        deployed.
-
-    Returns:
-        Deployment
-    """
-    ServeUsageTag.API_VERSION.record("v1")
-    return _private_api.get_deployment(name)
+    raise ValueError("serve.get_deployment is fully deprecated.")
```
Comment (shrekris-anyscale) [inline python/ray/serve/api.py:429]:
```suggestion
    raise ValueError("serve.get_deployment() is fully deprecated. Use serve.get_app_handle() to get a handle to a running Serve application.")
```

File: python/ray/serve/api.py [hunk 2]
```diff
@@ -426,37 +426,13 @@ def decorator(_func_or_class):
...
 @guarded_deprecation_warning(instructions=MIGRATION_MESSAGE)
 @Deprecated(message=MIGRATION_MESSAGE)
 def list_deployments() -> Dict[str, Deployment]:
-    """Returns a dictionary of all active deployments.
-
-    Dictionary maps deployment name to Deployment objects.
-    """
-    ServeUsageTag.API_VERSION.record("v1")
-    return _private_api.list_deployments()
+    raise ValueError("serve.list_deployments() is fully deprecated.")
```
Comment (shrekris-anyscale) [inline python/ray/serve/api.py:435]:
```suggestion
    raise ValueError("serve.list_deployments() is fully deprecated. Use serve.status() to get a list of all active deployments.")
```

File: python/ray/serve/api.py [hunk 3]
```diff
@@ -426,37 +426,13 @@ def decorator(_func_or_class):
 @guarded_deprecation_warning(instructions=MIGRATION_MESSAGE)
 @Deprecated(message=MIGRATION_MESSAGE)
```
Comment (shrekris-anyscale) [inline python/ray/serve/api.py:427]: Do we still need these deprecation messages, or is raising the error enough?

**Review by shrekris-anyscale — APPROVED** [review]
Nice work! I left a few nits.

**Comment by zcin** [comment]
@edoakes Tests are passing!

**Comment by zcin** [comment]
Hi @ericl @iycheng @stephanie-wang could I get a review/stamp on the changes in the workflow test files?

**Review by angelinalg — APPROVED** [review]

**Review by stephanie-wang — APPROVED** [review]
Approving workflow test changes.

**Pull request merged**